src/sentry/snuba/discover.py

```diff
@@ -482,7 +482,7 @@ def top_events_timeseries(
             )
             if values:
                 # timestamp needs special handling, creating a big OR instead
-                if field == "timestamp":
+                if field in ["timestamp", "timestamp.to_hour", "timestamp.to_day"]:
```

Comment:
This will add a condition for `timestamp` instead of `timestamp.to_hour`, and this if condition will never be reached because `timestamp.to_hour` is caught by L473 already

Reply:
yes but that's `.alias` so actually it will be hit but also actually it will add the wrong condition because the next line is "timestamp". i tested it and it stops the crash but it also doesn't do the right thing.